### DISCUSSION TITLE
In SB6, why does using `react-docgen-typescript` default to .mjs/.jsx extensions but not .tsx?

### DISCUSSION THREAD

**Discussion by brandonheyer** [storybookjs/storybook#12537] [Help]
As the title says, any idea why the default setting for `reactDocgen = 'react-cogent-typescript'` excludes .tsx files as seen here:

https://github.com/storybookjs/storybook/blob/ea341b594cddc3162cfadf703a235f609cf48031/app/react/src/server/framework-preset-react-docgen.ts#L17

That seems like a huge oversight, if I change the config manually via:
```
// webpack.config.js
config.module.rules[0].use[0].options.overrides[0].test = /\.(tsx)$/;
```

It pulls in the description of the .tsx component correctly. Any insight into this decision would be great, I want to make sure I'm leveraging as much default behavior as possible.

**Comment by shilman** [comment]
No bug here AFAICT.

This line says that if you've set `reactDocgen` to `react-docgen` then DON'T apply `babel-plugin-react-docgen` to typescript files. Why not? Because `react-docgen-typescript-plugin` will be applied to those files.

It's possible what you're seeing is some bug in `react-docgen-typescript` that is fixed in `react-docgen`.
